gef-ui/server/controllers/companies-house/index.js

```diff
@@ -0,0 +1,64 @@
+import _isEmpty from 'lodash/isEmpty';
+import * as api from '../../services/api';
+import { validationErrorHandler } from '../../utils/helpers';
+
+const companiesHouse = async (req, res) => {
+  const { params } = req;
+  const { applicationId } = params;
+
+  try {
+    const { exporterId } = await api.getApplication(applicationId);
+    const { details } = await api.getExporter(exporterId);
+    const { companiesHouseRegistrationNumber } = details;
+
+    return res.render('partials/companies-house.njk', {
+      regNumber: companiesHouseRegistrationNumber,
+      applicationId,
+    });
+  } catch (err) {
+    return res.render('partials/problem-with-service.njk');
+  }
+};
+
+const validateCompaniesHouse = async (req, res) => {
+  const { params, body } = req;
+  const { regNumber } = body;
+  const { applicationId } = params;
+
+  try {
+    const { exporterId } = await api.getApplication(applicationId);
+
+    if (_isEmpty(regNumber)) {
+      const regNumberError = {
+        errRef: 'regNumber',
+        errMsg: 'Enter a Companies House registration number',
+      };
+
+      return res.render('partials/companies-house.njk', {
+        errors: validationErrorHandler(regNumberError),
+        regNumber,
+        applicationId,
+      });
+    }
+
+    await api.getCompaniesHouseDetails(regNumber, exporterId);
+
+    return res.redirect('exporters-address');
+  } catch (err) {
+    // Validation errors
+    if (err.status === 422) {
```

Comment:
is the 422 status check needed?

Reply:
To keep errors consistent throughout the whole application, we made all **validation** errors have a status code of 422